util/download_util_script.py

```diff
@@ -0,0 +1,236 @@
+
+    Returns:
+        bool: True if the download (and optional unzip) was successful, False otherwise.
+    """
+    logging.info(f"Attempting to download from: {url}")
+    file_path = None  # Initialize file_path for cleanup in except block
+
+    try:
+        # --- URL scheme validation: Crucial for handling MissingSchema error ---
+        parsed_url = urlparse(url)
+        if not parsed_url.scheme or not parsed_url.netloc:
+            logging.error(f"Invalid URL format or missing scheme for '{url}'. "
+                          "Please ensure URL starts with 'http://' or 'https://'.")
+            return False
+        # --- End URL scheme validation ---
+
+        # --- Output folder validation: Ensure it's not an empty string ---
+        if not output_folder or not output_folder.strip(): # .strip() handles strings with only spaces
+            logging.fatal(f"Invalid output_folder specified: '{output_folder}'. "
+                          "Output path cannot be empty or consist only of whitespace.")
+            return False
+        # --- End Output folder validation ---
+
+        file_name = os.path.basename(parsed_url.path)
+        if not file_name:
+            logging.warning(f"Unable to infer filename from URL '{url}'. Using a default name.")
+            file_name = "downloaded_file"
+
+        # Logic to append .xlsx if no extension is found and it's not a zip
+        if '.' not in file_name and not unzip:
+            file_name = file_name + '.xlsx'
+            logging.info(f"Appended '.xlsx' extension. Final inferred filename: {file_name}")
+
+        file_path = os.path.join(output_folder, file_name)
+
+        # Ensure output folder exists
+        os.makedirs(output_folder, exist_ok=True)
+        logging.info(f"Output folder '{output_folder}' ensured to exist.")
+
+        # If file already exists, skip download
+        if os.path.exists(file_path):
+            logging.info(f"File already exists at {file_path}. Skipping download.")
+            if unzip and file_name.endswith('.zip'):
+                try:
+                    unzip_file(file_path, output_folder)
+                    logging.info("Existing zip file successfully unzipped.")
+                except Exception as e:
+                    logging.error(f"Failed to unzip existing file {file_path}: {e}")
+                    return False
+            return True # Explicit return True on existing file skip (or successful unzip)
+
+        response = _retry_method(url, headers, tries, delay, backoff)
+
+        # Stream content to file to handle potentially large files
+        with open(file_path, "wb") as f:
+            for chunk in response.iter_content(chunk_size=8192):  # 8KB chunks
+                f.write(chunk)
+
+        logging.info(f"File successfully downloaded to: {file_path}")
+
+        if unzip and file_name.endswith('.zip'):
+            try:
+                unzip_file(file_path, output_folder)
```

Comment:
Can the exception handling be moved within unzip_file() so this block of code is not repeated from line 155 above?